### PR TITLE
fix: manual depr entry not updating asset value [v14]

### PR DESCRIPTION
**Pull request by anandbaburajan**
1. When manual depreciation entries (created using the `Create Depreciation Entry` option in the `Manage` dropdown of assets with `calculate_depreciation` not checked) are submitted, the value of the asset isn't updated, so fixed that.
2. We have quite a few functions which (sometimes incorrectly) performed the same task of fetching an asset's current value after depreciation, so refactored all of them and now there's just one function.
3. The fixed asset register and the chart on asset documents weren't reflecting manual depr entries, so fixed them.
4. Wrote a patch to update value of all assets which have manual depr entries created for them.
5. In the fixed asset report register, assets with finance books weren't filtered properly resulting in an `TypeError: unsupported operand type(s) for +=: 'int' and 'NoneType'` error, so fixed it.

### CODE REVIEW

File: erpnext/accounts/doctype/journal_entry/journal_entry.py [hunk 1]
```diff
@@ -225,6 +226,32 @@ def apply_tax_withholding(self):
...
+
+		processed_assets = []
+
+		for d in self.get("accounts"):
+			if (
+				d.reference_type == "Asset" and d.reference_name and d.reference_name not in processed_assets
+			):
+				processed_assets.append(d.reference_name)
+
+				asset = frappe.get_doc("Asset", d.reference_name)
```
Comment (deepeshgarg007) [inline erpnext/accounts/doctype/journal_entry/journal_entry.py]: Use `get_value` instead of `get_doc`

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

**Comment by barredterra** [comment]
Is there a corresponding PR that implements these changes on the develop branch?

**Comment by anandbaburajan** [comment]
@barredterra not yet, I've to think more about some things. I'll open a PR in a day or two.

**Comment by barredterra** [comment]
@AnandBaburajan isn't the correct procedure to send PRs to develop, then (partially) backport them? This way we can avoid features going missing with an upgrade. For example, if somebody upgrades to develop right now, your features will not be there.

**Comment by anandbaburajan** [comment]
> @AnandBaburajan isn't the correct procedure to send PRs to develop, then (partially) backport them?

Yep, but like I said, I've to think more about some of those features/fixes (there might be some design changes) which would go into develop and fixing bugs in v13 and v14 was the first priority so opened this PR.

**Comment by barredterra** [comment]
Understood. IMHO it doesn't make sense to solve a bug in a lower version and "maybe sometime" (to exaggerate) in the `develop` branch. This has lead to problems in the past and will continue to do so, unless we follow the proper process.

**Pull request merged**